goagen/gen_client/generator.go

```diff
@@ -990,6 +990,7 @@ func (c *Client) {{ $funcName }}(ctx context.Context, path string{{ if .Params }
 		return nil, err
 	}
 {{ if or .Headers .HasPayload }}	header := req.Header
+		_ = header // Avoid "header declared and not used" error
 {{ if .HasPayload }}{{ if .HasMultiContent }}	if contentType != "*/*" {
```

Comment:
It seems moving the check for `HasMultiContent` might be better, something like:
```go
{{ if or .Headers (and .HasPayload .HasMultiContent) }}	header := req.Header
{{ if and .HasPayload .HasMultiContent }}	if contentType != "*/*" {
```
The idea being that we only to set headers if there's a payload and a `contentType` parameter or if there are headers to be set. The code above also combines the two `if` on line `994` into one which means one of the `end` below (on line `997` must be removed). Thank you!